### PR TITLE
Add compatibility with GM4_Resources

### PR DESCRIPTION
**Pull request by misode**

### CODE REVIEW

**Comment by AndanteDevs** [comment]
Will close #174

**Comment by AndanteDevs** [comment]
**Note** *Will close #174*

**Comment by misode** [comment]
This pull request modifies items (custom items, advancement icons, as a model on an armor stand) and custom entities. They now all have a `CustomModelData` if applicable and all text uses translation keys, if possible using the vanilla keys. See [GM4_Resources](https://github.com/Gamemode4Dev/GM4_Resources) for more info

Some modules were ignore because updates were still required: sunken treasure (#184) and trapped signs (no PR yet)

File: gm4_template_pack/data/gm4/advancements/MODULE_ID.json [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
     },
-    "title":"Plain Ol' Advancement Title",
-    "description":{
-      "text":"Noone's bothered to give me a description yet... They won't... They don't care about me....",
-      "color":"gray"
+    "title": {
+      "translate": "%1$s",
+      "with": [
+        "Plain Ol' Advancement Title",
+        {"translate": "advancement.gm4.template_pack.title"}
```
Comment (AndanteDevs) [inline gm4_template_pack/data/gm4/advancements/MODULE_ID.json]: In theory, shouldn't this and its description be `advancement.gm4.MODULE_ID` not `advancement.gm4.template_pack`? It just helps with a mass replace when using the template

Comment (AndanteDevs) [reply gm4_template_pack/data/gm4/advancements/MODULE_ID.json]: I just realised, in the resource pack you have the language string for the template pack, but no-one sees these dummy strings anyways?

Comment (misode) [reply gm4_template_pack/data/gm4/advancements/MODULE_ID.json]: All folders need to be lowercase so that wouldn't work

**Pull request merged**